onnxruntime/test/python/onnxruntime_test_python.py

```diff
@@ -43,6 +43,20 @@ def testSetProviders(self):
             # confirm only CPU Provider is registered now.
             self.assertEqual(['CPUExecutionProvider'], sess.get_providers())
 
+    def testSetProvidersWithOptions(self):
+        if 'CUDAExecutionProvider' in onnxrt.get_available_providers():
```

Comment:
@HectorSVC  , is there a good way to test/confirm which cuda device was used to run a session?

Reply:
https://gist.github.com/f0k/63a664160d016a491b2cbea15913d549
I found this approach to use python ctypes to call cuda library such as libcuda.so, so that we can get device info. 
I'll try to add this check to test it.